lib/ansible/modules/packaging/os/pacman.py

```diff
@@ -194,7 +194,7 @@ def upgrade(module, pacman_path):
     }
 
     if rc == 0:
-        regex = re.compile(r'([\w-]+) ((?:\S+)-(?:\S+)) -> ((?:\S+)-(?:\S+))')
+        regex = re.compile(r'([\w+\-.@]+) ((?:\S+)-(?:\S+)) -> ((?:\S+)-(?:\S+))')
```

Comment:
```suggestion
        # https://wiki.archlinux.org/index.php/Arch_package_guidelines#Package_naming
        # Package names can contain only alphanumeric characters and any of @, ., _, +, -.
        # All letters should be lowercase.
        regex = re.compile(r'([\w+-.@]+) ((?:\S+)-(?:\S+)) -> ((?:\S+)-(?:\S+))')
```

I think it should be useful to add some comment ; and to change `\-` to `-`  (why did you add a backslash before the hyphen ?)
What do you think ?

Reply:
> I think it should be useful to add some comment

Seems excessive to add the package naming guidelines directly in the code.  And the last item (`# All letters should be lowercase`) isn't reflected in the code.  (Nor is it reflected in package names -- there are packages on my system w/ capital letters.)

I agree that a comment makes sense, though. This is what I added:

```python
    if rc == 0:
        # Match lines of `pacman -Qu` output of the form:
        #   (package name) (before version-release) -> (after version-release)
        # e.g., "ansible 2.7.1-1 -> 2.7.2-1"
        regex = re.compile(r'([\w+\-.@]+) ((?:\S+)-(?:\S+)) -> ((?:\S+)-(?:\S+))')
```

> and to change `\-` to `-` (why did you add a backslash before the hyphen ?)

An unescaped hyphen (`-` without `\`) inside a set `[]` indicates a range, except when it's the first or last character in the set.

The old regex didn't need to have the hyphen escaped, because it was the last character. For the updated one (w/ `+`, `.`, and `@`):

`[\w+-.@]` (without the `\`) matches any character in:
- `\w` (roughly `r'[A-Za-z0-9_]'`)
- or `+-.` the range of characters `+` through `.`
- or `@`

`[\w+\-.@]` (with the `\`) matches any character in:
- `\w` (roughly `r'[A-Za-z0-9_]'`)
- or `+`
- or `\-` (only matches `-`; `\` just escapes its "range" semantics)
- or `.`
- or `@`

Reply:
I also simplified the existing regex slightly by removing the unneeded non-capturing groups: `((?:\S+)-(?:\S+))` is exactly the same as `(\S+-\S+)`.

Reply:
> Seems excessive to add the package naming guidelines directly in the code.
I personally like to have a link pointing to documentation ; but that's a personal taste.

I am totally ok with the comment you added.

> An unescaped hyphen (- without \) inside a set [] indicates a range, except when it's the first or last character in the set.
> The old regex didn't need to have the hyphen escaped, because it was the last character.
Very well explained, thanks !